File: 前缀和/523.py
from typing import List


class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        if len(nums) < 2:
            return False
        dic = {0:-1}
        mod = 0
        for i in range(len(nums)):
            mod = (mod+nums[i])%k
            idx = dic.get(mod)
            if idx is not None:
                if (i-idx)>=2:
                    return True
                else:
                    continue
            dic[mod] = i
        return False


nums=[23,24,7,6]
k=31
obj=Solution()
print(obj.checkSubarraySum(nums, k))

File: 前缀和/523.cpp
#include<vector>
#include<unordered_map>
using namespace std;
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        if(nums.size()<2)
            return false;
        unordered_map<int,int>dict;
        dict[0] = -1;
        int mod = 0;
        for(int i=0;i<nums.size();i++){
            mod += nums[i];
            mod = mod%k;
            if(dict.find(mod)!=dict.end()){
                if((i-dict[mod])>=2)
                    return true;
            }
            else
                dict[mod] = i;
        }
        return false;
    }
};